src/Symfony/Component/Lock/Store/StoreFactory.php

```diff
@@ -25,17 +25,8 @@
  */
 class StoreFactory
 {
-    /**
-     * @param \Redis|\RedisArray|\RedisCluster|\Predis\ClientInterface|RedisProxy|RedisClusterProxy|\Memcached|\MongoDB\Collection|\PDO|Connection|\Zookeeper|string $connection Connection or DSN or Store short name
-     *
-     * @return PersistingStoreInterface
```

Comment:
not sure we can add the real return type here

Reply:
Revert. Why can't we do that?

Reply:
adding a return type breaks child classes
we first need a deprecation layer in 5.4